### ISSUE TITLE
NS2002 falsely triggered when substituting an interface

### ISSUE THREAD

**Issue by davidkeaveny** [nsubstitute/NSubstitute.Analyzers#205]
I have a simple interface and a class that dispatches against it:

```
public interface IReportRunner
{
  void Execute(int year, int month);
}

public class ClassThatTakesInterfaceDependency
{
  private readonly IReportRunner _runner;

  public ClassThatTakesInterfaceDependency(IReportRunner runner)
  {
    _runner = runner;
  }

  public void DoStuffThatInvokesInterface(DateTime now)
  {
    _runner.Execute(now.Year, now.Month);
  }
}
```

When I create a test that exercises the interface, I'm getting an NS2002 error when creating the substitute:

```
public class RunMyTest
{
  [Fact]
  public void ItShouldExecute()
  {
    // Arrange
    var substitute = Substitute.For<IReportRunner>();
    var sut = new ClassThatTakesInterfaceDependency(substitute);
    
    // Act
    sut.DoStuffThatInvokesInterface(new DateTime(2023, 6, 5));

    // Assert
    substitute.Received(1).Execute(2023, 6);
  }
}
```

Given that it is substituting an interface, it shouldn't be complaining about constructor parameters.

I am using NSubstitute 5.0.0 with NSubstitute.Analyzers 1.0.16.

**Comment by tpodolak** [comment]
hi @davidkeaveny 
I cannot repro this - no warnings reported both when testing with IDE and with integration tests
![image](https://github.com/nsubstitute/NSubstitute.Analyzers/assets/7378346/9a1de66c-76c0-4bd5-b03c-ed28f1f2ba9f)

**Comment by DrunkRussianGun** [comment]
Hello
I sometimes run into the same problem. That's a heisenbug so it can be difficult to reproduce even with the same code example, but it appears regularly.

**Comment by SonicGD** [comment]
Hello. Can confirm. Sometimes these errors just appears. Only Rider cache invalidation helps.

**Comment by tpodolak** [comment]
There is not much I can do here. This seems to be Rider Roslyn analyzers support issue rather than NSubstitute.Analyzers bug. I will send bug report to Jetbrains if I encounter this problem on my own - as reporting this without any logs doesnt make sense. Alternatively @SonicGD or @davidkeaveny you can report this on your own https://youtrack.jetbrains.com/issues/RIDER and link the issue here

**Comment by askvira-woolworthslimited** [comment]
My 2 cents. 

Had this issue when mocking a source-generator-produced interface (using HotChocolate source gens).

Solved, or rather, side-stepped this issue by creating a sham interface that inherits the problematic one, and mocking that sham interface instead.

**Comment by StaticSphere** [comment]
Issue exists in July of 2025, using .NET 9. I'm guessing it's a Rider issue from what I'm reading here and from invalidating the cache. But I can confirm that invaliding Rider's cache and restarting does fix the issue.
